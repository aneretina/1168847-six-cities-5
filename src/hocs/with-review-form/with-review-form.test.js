import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withReviewForm from "./with-review-form";


const noop = () => {};
const review = `Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`;
const rating = `2`;

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withReviewForm(MockComponent);

describe(`WithForm render`, () => {
  it(`withForm is rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        rating={rating}
        review={review}
        onRatingChange={noop}
        onTextChange={noop}
        resetState={noop}
        isDisabled={true}
        updateDisableFormStatus={noop}>
        <React.Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
