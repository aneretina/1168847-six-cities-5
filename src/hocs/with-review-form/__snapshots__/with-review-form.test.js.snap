// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithForm render withForm is rendered correctly 1`] = `<div />`;
